gio-subclass/tests/simple_application.rs

```diff
@@ -0,0 +1,189 @@
+                if line.starts_with("--local-"){
+                    println!("handling argument {:?} locally", line);
+                    rm.push(i)
+                }else{
+                    println!("not handling argument {:?} locally", line);
+
+                }
+            }
+
+            rm.reverse();
+
+            for i in rm.iter(){
+                arguments.remove(*i);
+            }
+
+            None
+        }
+
+        fn command_line(&self, application: &Application, cmd_line: &gio::ApplicationCommandLine) -> i32 {
+
+            let arguments = cmd_line.get_arguments();
+
+            for arg in arguments{
+                println!("handling argument {:?} remotely", arg);
+            }
+
+            return 0;
+        }
+
+    }
+
+    struct SimpleApplicationStatic;
+
+    impl ImplTypeStatic<Application> for SimpleApplicationStatic {
+        fn get_name(&self) -> &str {
+            "SimpleApplication"
+        }
+
+        fn new(&self, obj: &Application) -> Box<ApplicationImpl<Application>> {
+            SimpleApplication::init(obj)
+        }
+
+        fn class_init(&self, klass: &mut ApplicationClass) {
+            SimpleApplication::class_init(klass);
+        }
+    }
+}
+
+glib_wrapper! {
+    //TODO: is this correct???
+    pub struct SimpleApplication(Object<imp::SimpleApplication>):
+        [Application => InstanceStruct<Application>,
+         gio::Application => gio_ffi::GApplication];
+
+
+    match fn {
+        get_type => || imp::SimpleApplication::get_type().to_glib(),
+    }
+}
+
+impl SimpleApplication {
+
+    pub fn new<'a, I: Into<Option<&'a str>>>(application_id: I, flags: gio::ApplicationFlags) -> Result<SimpleApplication, glib::BoolError> {
+        use glib::object::Downcast;
+
+        unsafe {
+            match glib::Object::new(Self::static_type(), &[("application-id", &application_id.into()),
+                                                           ("flags", &flags)]){
+                Ok(obj) => Ok(obj.downcast_unchecked()),
+                Err(_) => Err(glib::BoolError("Failed to create application"))
+            }
+        }
+    }
+
+}
+
+// TODO: This one should probably get a macro
+impl ops::Deref for SimpleApplication {
+    type Target = imp::SimpleApplication;
+
+    fn deref(&self) -> &Self::Target {
+        unsafe {
+            let base: Application = from_glib_borrow(self.to_glib_none().0);
+            let imp = base.get_impl();
+            let imp = imp.downcast_ref::<imp::SimpleApplication>().unwrap();
+            // Cast to a raw pointer to get us an appropriate lifetime: the compiler
+            // can't know that the lifetime of base is the same as the one of self
+            &*(imp as *const imp::SimpleApplication)
+        }
+    }
+}
+
+#[test]
+fn test_arguments() {
+    let app = SimpleApplication::new("org.gtk.TestApplication", gio::ApplicationFlags::empty()).unwrap();
+
+    app.set_inactivity_timeout(10000);
```

Comment:
Does `run()` below block for 10s now? That's suboptimal for a test :) It would be better to check if all is running fine, and then explicitly stop the application directly... as that would then run as fast as possible.

Reply:
Not at all. I don't have a clue why that line is there. Again, just copied from the original test, since I don't know how to use that vfunc.

Reply:
Ok, let's just keep it like this then for now but please add a FIXME comment above